solana/automata-dcap-framework/sdk/src/verifier_client.rs

```diff
@@ -0,0 +1,433 @@
+                },
+                other => return Err(anyhow::anyhow!("Unsupported issuer common name: {}", other)),
+            }
+
+            let pk: VerifyingKey= (issuer)
+                .try_into()
+                .map_err(|e| anyhow::anyhow!("failed to decode key from certificate: {}", e))?;
+
+            pk.verify_strict(cert)
+                .map_err(|e| anyhow::anyhow!("failed to verify certificate: {}, error: {}", cert.tbs_certificate.subject.to_string(), e))?;
+        }
+        Ok(pck_cert_chain_data.pck_extension.fmspc)
+
+    }
+
+    async fn verify_tcb_status(&self, quote: &Quote<'_>) -> anyhow::Result<(TcbStatus, Vec<String>)> {
+        let pck_extension = quote.signature.get_pck_extension()?;
+        let fmspc = hex::encode(pck_extension.fmspc);
+        let version = 3 as u8;
+        let tcb_type = if quote.header.tee_type == SGX_TEE_TYPE {
+            TcbType::Sgx
+        } else {
+            TcbType::Tdx
+        };
+        let tcb_info_data = self.pccs_client.get_tcb_info_data(tcb_type, fmspc, version).await?;
+        let tcb_info: TcbInfoAndSignature = serde_json::from_slice(&tcb_info_data)?;
+        let tcb_info = tcb_info.get_tcb_info()?;
+
+        let (mut tcb_status, advisory_ids) = verify_tcb_status(&tcb_info, &pck_extension)?;
+        if quote.header.tee_type == TDX_TEE_TYPE {
+            let tdx_module_status = tcb_info.verify_tdx_module(quote.body.as_tdx_report_body().unwrap())?;
+            tcb_status = TcbInfo::converge_tcb_status_with_tdx_module(tcb_status, tdx_module_status);
+        }
+
+        Ok((tcb_status, advisory_ids))
+    }
+
+    async fn update_verified_output_account(
+        &self,
+        quote_buffer_pubkey: Pubkey,
+        verified_output_pubkey: Pubkey,
+        mut tcb_status: TcbStatus,
+        advisory_ids: Vec<String>,
+        fmspc: [u8; 6]
+    ) -> anyhow::Result<Signature> {
+
+        let qe_tcb_status = self.get_qe_tcb_status(quote_buffer_pubkey).await?;
+        let qe_tcb_status = QeTcbStatus::from_str(&qe_tcb_status.status).map_err(
+            |e| anyhow::anyhow!("failed to parse qe tcb status: {}", e)
+        )?;
+
+        tcb_status = TcbInfo::converge_tcb_status_with_qe_tcb(tcb_status, qe_tcb_status.into());
+
+
+        let tx = self.program.request()
+            .accounts(accounts::UpdateVerifiedOutputAccount {
+                owner: self.program.payer(),
+                verified_output: verified_output_pubkey,
+                quote_data_buffer: quote_buffer_pubkey,
+                system_program: anchor_client::solana_sdk::system_program::ID,
+            })
+            .args(args::UpdateVerifiedOutputAccount {
+                tcb_status: tcb_status.to_string(),
+                advisory_ids,
+                fmspc,
+            })
+            .send().await?;
+
+        Ok(tx)
+    }
+
+
+    async fn check_certificate_revocation(
```

Comment:
this needs to be moved onchain as well. we will do this as part of the ZK/onchain ECDSA verification

Reply:
not within scope for now. we can make a new PR for this. I don't want this PR to be too chonky.